File: todolist-back-python/primary/controller/write/choose_and_ignore_task.py
from uuid import UUID

from hexagon.fvp.type import TaskKey
from hexagon.fvp.write.choose_and_ignore_task import ChooseAndIgnoreTaskFvp
from primary.controller.dependency_list import DependencyList


def choose_and_ignore_task(chosen_task: UUID, ignored_task: UUID, dependencies: DependencyList) -> None:
    ChooseAndIgnoreTaskFvp(dependencies.fvp_session_repository_for_fvp()).execute(TaskKey(chosen_task), TaskKey(ignored_task))

File: todolist-front-ts/primary/controller/chooseAndIgnoreTask.ts
import {WhichTaskUpdated} from "@/lib/todolist.slice";
import {Dependencies} from "@/primary/controller/dependencies";
import {WhichTask} from "@/hexagon/whichTaskQuery/whichTask.query";
import {ChooseAndIgnoreTask} from "@/hexagon/chooseTask/chooseTask.usecase";

export class Controller {
    private _store: StoreContract;
    private _whichTaskQuery: WhichTask.Contract;
    private _chooseAndIgnoreTask: ChooseAndIgnoreTask.Contract;

    constructor(dependencies: Dependencies, store: StoreContract) {
        this._store = store;
        this._whichTaskQuery = dependencies.whichTask.useCase(dependencies);
        this._chooseAndIgnoreTask = dependencies.chooseAndIgnoreTask.useCase(dependencies);
    }

    async chooseAndIgnoreTask(chosenTaskId: number, ignoredTaskId: number) {
        await this._chooseAndIgnoreTask.execute(chosenTaskId, ignoredTaskId);
        const tasks = await this._whichTaskQuery.query();
        this._store.dispatch(WhichTaskUpdated({tasks: tasks}));
    }
}

export interface StoreContract {
    dispatch(whichTaskUpdated: any): void;
}
